room.go

```diff
@@ -457,26 +457,28 @@ func (r *Room) handleResumed() {
 	r.sendSyncState()
 }
 
-func (r *Room) handleDataReceived(identity string, dataPacket DataPacket) {
-	if identity == r.LocalParticipant.Identity() {
+func (r *Room) handleDataReceived(senderIdentity string, dataPacket DataPacket) {
+	if senderIdentity == r.LocalParticipant.Identity() {
 		// if sent by itself, do not handle data
 		return
 	}
-	p := r.GetParticipantByIdentity(identity)
 	params := DataReceiveParams{
-		SenderIdentity: identity,
-		Sender:         p,
+		SenderIdentity: senderIdentity,
+		Sender:         r.GetParticipantByIdentity(senderIdentity),
 	}
+	// only local participant should receive data callbacks
```

Comment:
This seems like an API change. Is there any reason? Has it been discussed?

Reply:
I may be misunderstanding it, but when user connect with SDK he can set two data callback: on the room and on the (local) participant. And the second callback doesn't fire when data message is received. I'm not sure why it was set to fire on a _remote_ participant instead of the local. Looks like a bug to me.

Reply:
Is it meant to fire on the participant object representing the metadata sender or the receiver?

Reply:
it's meant to be fired on the sender. docs here: https://github.com/livekit/client-sdk-js/blob/main/src/room/events.ts#L424

Reply:
That's pretty confusing, I must admit.

Why do we have `OnDataPacket` on the `LocalParticipant` then? If the logic is "fire at the sender", local participant [will always skip](https://github.com/livekit/server-sdk-go/pull/499/commits/ed06c90dca2971b504ec44bff439772589dab4a5#diff-49f4dcd9caec275f255f290d41ae11cc0e47384c4f45c9ff2256bf80ec639e12R461) their own messages.

Reply:
I'm not sure, but reach out more widely to folks who have been maintaining that piece of code to understand the intended behavior.

Reply:
yeah.. makes no sense on the `LocalParticipant`. There are two ways of receiving data:
1. give me all data messages (listen on room)
2. give me all messages sent from a specific remote participant (listen on RemoteParticipant)